lib/auth/github_authenticator.rb

```diff
@@ -113,4 +117,15 @@ def register_middleware(omniauth)
            },
            scope: "user:email"
   end
+
+  protected
+
+  def retrieve_avatar(user, data)
+    return unless user
+    return if user.user_avatar&.custom_upload_id.present?
+
+    if (avatar_url = data[:image]).present?
+      Jobs.enqueue(:download_avatar_from_url, url: avatar_url, user_id: user.id, override_gravatar: false)
```

Comment:
`avatar_url` will be something like `https://avatars0.githubusercontent.com/u/6376558?v=4`. 

It has an optional parameter size `s`, with a default and maximum at 460px x 460px.